coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala

```diff
@@ -39,38 +39,58 @@ class SinglePartitionPlanner(planners: Map[String, QueryPlanner],
     if(planner.isEmpty)  planners.values.head else planner.head
   }
 
-  private def materializeSimpleQuery(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
-    getPlanner(logicalPlan).materialize(logicalPlan, qContext)
-  }
+  private def getPlanner(binaryJoin: BinaryJoin) : Seq[QueryPlanner] = {
```

Comment:
Can you explain this function in plain english ? If lhs is not a binary join, then `getPlanner(lhs)` ... not understanding how this is compiling.

Reply:
I changed the overloaded `getPlanner` to `getBinaryJoinPlanners`. For BinaryJoin we need to call getPlanner on LHS and RHS and return the combined result.